index.d.ts

```diff
@@ -43,7 +43,7 @@ export interface Action {
  *
  * @template S State object type.
  */
-export type Reducer<S> = <A extends Action>(state: S | undefined, action: A) => S;
+export type Reducer<S, A extends Action = Action> = (state: S, action: A) => S;
```

Comment:
I don't think so.
For each `Reducer`, they may need to have the same structure of `State`, includes the initial state.

And we used to  define `Reducer` with defaults like:
```ts
const someReducer: Reducer<State> = (state = {} as State  /* this is may not undefined */, action) => state
```